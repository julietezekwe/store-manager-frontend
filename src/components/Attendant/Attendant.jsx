import React from 'react';

const Attendant = () => (
      <React.Fragment>

        <div className="dashboard myCard col-2">
          <h3> Dashboard</h3>
          <p className="active"><a href="/attendant/">Products</a></p>
          <hr />
          <p className="active"><a href="/attendant/sales">Sales Records</a></p>

        </div>
      </React.Fragment>
);

export default Attendant;
