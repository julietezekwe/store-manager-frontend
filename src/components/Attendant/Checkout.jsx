/* eslint-disable no-restricted-globals */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getAllCartProducts, handleCheckout } from '../../actions/salesAction';

export class Checkout extends Component {
  state = { totalPrice: 0 };

  async componentDidMount() {
    if (JSON.parse(localStorage.getItem(`${this.props.login.user.id}`))) {
      const carts = JSON.parse(
        localStorage.getItem(`${this.props.login.user.id}`),
      );
      await this.props.getAllCartProducts(carts.cartNo);
    }
  }

  handleRemoveFromCart = (id) => {
    const carts = JSON.parse(
      localStorage.getItem(`${this.props.login.user.id}`),
    );

    for (let i = 0; i < carts.cartNo.length; i += 1) {
      if (Number(carts.cartNo[i]) === Number(id)) {
        carts.cartNo.splice(i, 1);
      }
    }
    const cartContent = JSON.stringify(carts);
    localStorage.setItem(`${this.props.login.user.id}`, cartContent);
    localStorage.setItem('cartCount', carts.cartNo.length);
    window.location.reload();
  };

  onQuantityChange = (quantity, id, price) => {
    if (quantity < 1 || isNaN(quantity)) quantity = 0;
    document.getElementById(`Q${id}`).innerHTML = quantity;
    const currentPrice = document.getElementById(`${id}`).innerHTML;
    const newPrice = quantity * price;

    document.getElementById(`${id}`).innerHTML = newPrice;
    let totalPrice = Number(document.getElementById('totalPrice').innerHTML);
    totalPrice = totalPrice - currentPrice + newPrice;
    this.setState({ totalPrice });
  };

  render() {
    const { cart } = this.props.sales;

    return (
      <div className="col-8">
        <div
          className="card col-12"
          style={{
            float: 'none',
            textAlign: 'center',
            marginLeft: 'auto',
            marginRight: 'auto',
            marginTop: '35px',
          }}
        >
          <div className="card col-9" id="alert" />
          <table className="table" style={{ width: '100%' }} id="checkout">
            <thead>
              <tr>
                <th>S/N</th>
                <th>Product Name</th>
                <th>Description</th>
                <th>ID</th>
                <th />
                <th>Quantity</th>
                <th>Amount (&#8358;)</th>
                <th>Action</th>
              </tr>
            </thead>
            <tbody>
              {cart
                && cart.cartProductDetails.map(product => (
                    <tr key={product[0].id}>
                      <td>{product[0].id}</td>
                      <td>{product[0].productname}</td>
                      <td>{product[0].description}</td>
                      <td>{product[0].id}</td>
                      <td>
                        <input
                          type="number"
                          min="1"
                          style={{ textAlign: 'center', width: '50px' }}
                          onChange={e => this.onQuantityChange(
                            e.target.value,
                            product[0].id,
                            product[0].price,
                          )
                          }
                        />{' '}
                      </td>
                      <td id={`Q${product[0].id}`}>1</td>
                      <td id={product[0].id}>{product[0].price}</td>
                      <td>
                        <i
                          className="far fa-trash-alt trigger"
                          onClick={() => this.handleRemoveFromCart(product[0].id)
                          }
                        />
                      </td>
                    </tr>
                ))}
            </tbody>
            <tfoot>
              <tr>
                <td colSpan="5" />
                <td>Total Price </td>
                <td id="totalPrice">
                  {this.state.totalPrice !== 0
                    ? this.state.totalPrice
                    : cart.totalPrice}
                </td>
              </tr>
              <tr>
                <td colSpan="2">
                  <button
                    className="button"
                    onClick={() => this.props.handleCheckout(this.props.login.user.id)
                    }
                  >
                    Checkout Now
                  </button>
                </td>
                <td colSpan="4" />
                <td colSpan="2" />
              </tr>
            </tfoot>
          </table>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  login: state.login,
  sales: state.sales,
});
export default connect(
  mapStateToProps,
  { getAllCartProducts, handleCheckout },
)(Checkout);
