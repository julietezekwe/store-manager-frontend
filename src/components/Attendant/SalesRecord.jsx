import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getPersonalSales } from '../../actions/salesAction';

export class SalesRecord extends Component {
  async componentDidMount() {
    await this.props.getPersonalSales();
  }

  render() {
    const { mySales } = this.props.sales;
    return (
      <React.Fragment>
        <div className="col-8">
          <div className="center">
            <div className="col-3 " id="totalProducts">
              Total Products Sold:{' '}
              {mySales.records && mySales.records.totalProducts}
            </div>
            <div className="col-4" id="totalPrice">
              Amount Realized: &#8358;{' '}
              {mySales.records && mySales.records.amount}
            </div>
            <div className="col-3" id="totalSales">
              Number of sales: {mySales.records && mySales.records.totalSales}
            </div>
          </div>

          <div className="card col-11">
            <table className="table">
              <thead>
                <tr>
                  <th>Product Id</th>
                  <th>Product Name</th>
                  <th>Quantity</th>
                  <th>Price</th>
                  <th>Seller</th>
                  <th>Date</th>
                </tr>
              </thead>
              <tbody>
                {mySales.saleDetail
                  && mySales.saleDetail.map((sale, i) => {
                    const date = new Date(sale.created_at);
                    const dateString = new Date(
                      date.getTime() - date.getTimezoneOffset() * 60000,
                    )
                      .toISOString()
                      .split('T')[0];

                    return (
                      <tr key={i}>
                        <td>{sale.productid}</td>
                        <td>{sale.productname}</td>
                        <td>{sale.quantity}</td>
                        <td>{sale.price}</td>
                        <td>{this.props.user.name}</td>
                        <td>{dateString}</td>
                      </tr>
                    );
                  })}
              </tbody>
            </table>
          </div>
        </div>
      </React.Fragment>
    );
  }
}
const mapStateToProps = state => ({
  sales: state.sales,
  user: state.login.user,
});
export default connect(
  mapStateToProps,
  { getPersonalSales },
)(SalesRecord);
