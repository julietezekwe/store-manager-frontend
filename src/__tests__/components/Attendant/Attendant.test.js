/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import Attendant from '../../../components/Attendant/Attendant.jsx';

let wrapper;
beforeEach(() => {
  wrapper = shallow(<Attendant />);
});
describe('App Component', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should display three links on the dashboard', () => {
    expect(wrapper.find('a')).toHaveLength(2);
  });
  it('should have the right path links for the hrefs', () => {
    expect(wrapper.find('a').at(0).prop('href')).toEqual('/attendant/');
  });
});
