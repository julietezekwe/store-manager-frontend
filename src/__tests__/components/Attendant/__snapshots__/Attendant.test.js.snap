// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should match snapshot 1`] = `
<Fragment>
  <div
    className="dashboard myCard col-2"
  >
    <h3>
       Dashboard
    </h3>
    <p
      className="active"
    >
      <a
        href="/attendant/"
      >
        Products
      </a>
    </p>
    <hr />
    <p
      className="active"
    >
      <a
        href="/attendant/sales"
      >
        Sales Records
      </a>
    </p>
  </div>
</Fragment>
`;
